tools/cmake/cores/Cortex-M7F.cmake

```diff
@@ -16,10 +16,10 @@ elseif(${MBED_TOOLCHAIN} STREQUAL "ARM")
         "-mfloat-abi=hard"
     )
     list(APPEND asm_compile_options
-        "-mcpu=Cortex-M7.fp.sp"
+        "-mcpu=Cortex-M7"
```

Comment:
I can confirm that `asm_compile_option` should share the same options as `c_cxx_compile_options` because they are processed by `armclang`.
A [similar issue was encountered for Cortex-M33](https://github.com/ARMmbed/mbed-os/pull/13968/commits/53cc382a96a36d87b9b72d524920dfbb9dba5a7e).

Reply:
@jeromecoutant Can you do the same in this PR as we did for m33 ?

Reply:
Done